libxrdp/xrdp_iso.c

```diff
@@ -410,10 +453,17 @@ xrdp_iso_send_cc(struct xrdp_iso *self)
         }
         else
         {
+            flags = EXTENDED_CLIENT_DATA_SUPPORTED;
+
+            if (client_info->vmconnect)
+            {
+                /* NLA is handled by the host and not us. */
+                flags |= RESTRICTED_ADMIN_MODE_SUPPORTED;
+            }
+
             /* [MS-RDPBCGR] RDP_NEG_RSP */
```

Comment:
Shouldn't we only set `RESTRICTED_ADMIN_MODE_SUPPORTED` if the client sets `RESTRICTED_ADMIN_MODE_REQUESTED` in the connection request PDU?

Reply:
The client sends `RESTRICTED_ADMIN_MODE_REQUIRED` (emphasis on the "REQUIRED" part).

Whether the server supports RestrictedAdmin or not is only a matter of Windows version (Windows > 2008), at least according to [MS-RDPBCGR].

Reply:
I understand that, but at least for other values, it's common practice for the client (or server) to suggest the capability (e.g. fastpath support) and the server (or client) to respond with the chosen capability. Is that not the case here? Am I misunderstanding how restricted admin mode works?

Reply:
I didn't really look into it that much honestly, but I don't really see why a server can't advertise for RestrictedAdmin support at this point.

If a client requires RestrictedAdmin and server answers here it doesn't, it fails. But you could imagine that a client doesn't "require" it, but does it opportunistically if available, as suggested by the "Supported" name. In any case, it doesn't affect functionality.

I should probably boot up a real Windows to see what it advertises.